### ISSUE TITLE
Crash in Chunking

### ISSUE THREAD

**Issue by kornatzky** [alvin0/redis-model#6]
When you are getting results via chunking, and there are no results, this piece of code, from file RedisRepository.php line 248, crashes in array_merge

`public function guaranteedScan(string $keyPattern, int $take, int $cursor = 0, $keyResultRemaining = [])
    {
        $cursor = $cursor === 0 ? ((string) $cursor) : $cursor;
        $keys = $keyResultRemaining;

        do {
            list($cursor, $result) = $this->getConnection()->scan($cursor, [
                'match' => $this->getRedisPrefix() . $keyPattern,
                'count' => $take,
            ]);
            ray('before merge', $cursor, $result);
            $keys = array_merge($keys, $result);
            if (sizeof($keys) > $take) {
                break;
            }
        } while ($cursor != '0');`

because $result is null

**Comment by kornatzky** [comment]
I added the ray calls to understand what is happening. Both $cursor and $result are null.

The error I see is:

`TYPE ERROR  array_merge(): Argument #2 must be of type array, null given`

**Comment by alvin0** [comment]
Can you tell me why this error occurs? I need to recreate it!

**Comment by kornatzky** [comment]
Here is the call:

```
     Message::where('auction_id', $auction->id)->where('created_at', '>', $x_minutes_ago)

      ->chunk(100, function($messages) {

         

      });
```

Message is a RedisModel but there are no keys for it in Redis. We did not insert anything yet for Message

**Comment by kornatzky** [comment]
Crash in line 

`$keys = array_merge($keys, $result);`

file RedisRepository.php

**Comment by alvin0** [comment]
Thank you for notifying me of this error, I have fixed it in version 0.6. Please upgrade to version 0.6 to continue experiencing it.

Commit: https://github.com/alvin0/redis-model/commit/299dc0171c67f065c113c0fad37f727b10ea0d7c

**Comment by kornatzky** [comment]
Works. Thanks. Great. I close this.
